### ISSUE TITLE
Test Failure: Integration tests 

### ISSUE THREAD

**Issue by jakelandis** [elastic/logstash#8035]
Noticed on the 5.6 branch:
```
Starting Logstash: /home/travis/build/elastic/logstash/build/logstash-5.6.0-SNAPSHOT/bin/logstash []
Sending Logstash's logs to /tmp/logstash-settings-test-0c8d14092948ad15543f36a2aa39d62ca895e230a2d7e865c4f6c73a297e which is now configured via log4j2.properties
[2017-08-18T19:04:32,722][INFO ][logstash.modules.scaffold] Initializing module {:module_name=>"fb_apache", :directory=>"/home/travis/build/elastic/logstash/build/logstash-5.6.0-SNAPSHOT/modules/fb_apache/configuration"}
[2017-08-18T19:04:32,725][INFO ][logstash.modules.scaffold] Initializing module {:module_name=>"netflow", :directory=>"/home/travis/build/elastic/logstash/build/logstash-5.6.0-SNAPSHOT/modules/netflow/configuration"}
[2017-08-18T19:04:32,751][FATAL][logstash.runner          ] The given configuration is invalid. Reason: Expected one of #, input, filter, output at line 1, column 1 (byte 1) after 

....


Failures:
  1) Test Logstash instance whose default settings are overridden should exit when config test_and_exit is set
     Failure/Error: expect(@logstash_service.exit_code).to eq(0)
     
       expected: 0
            got: 1
     
       (compared using ==)
     # ./specs/settings_spec.rb:93:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
Finished in 11 minutes 0 seconds (files took 4.74 seconds to load)
33 examples, 1 failure
Failed examples:
rspec ./specs/settings_spec.rb:80 # Test Logstash instance whose default settings are overridden should exit when config test_and_exit is set
Randomized with seed 3768
```

**Comment by jakelandis** [comment]
This is happening because the log directory and configuration directory are the same.  When testing the configuration it trying to validate the log file as a configuration file. 

I believe that this is only happening recently since it is the netflow and fb_apache modules which are now in the log file, where before those, i believe the log file was empty.

**Comment by jakelandis** [comment]
Closing this since it was fixed on 5.6 (where it was noted).  If this re-appears, we may need to foward port the fix on #8036
